OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs

```diff
@@ -78,12 +78,13 @@ public virtual void BuildingComplete(Actor self)
 
 		public void PlayCustomAnimation(Actor self, string name, Action after = null)
 		{
-			DefaultAnimation.PlayThen(NormalizeSequence(self, name), () =>
-			{
-				DefaultAnimation.Play(NormalizeSequence(self, Info.Sequence));
-				if (after != null)
-					after();
-			});
+			if (DefaultAnimation.HasSequence(name))
+				DefaultAnimation.PlayThen(NormalizeSequence(self, name), () =>
```

Comment:
We want this to throw because the outside code has asked this low level method to do something bogus.  The chexk should be in the reload code instead.